virtualcluster/cmd/syncer/app/options/options.go

```diff
@@ -362,3 +369,19 @@ func getClientConfig(config componentbaseconfig.ClientConnectionConfiguration, m
 
 	return restConfig, nil
 }
+
+func DnsOptionsConvert(dnsoptions map[string]string) []corev1.PodDNSConfigOption {
+	var podDnsOptions []corev1.PodDNSConfigOption
+	podDnsOptions = make([]corev1.PodDNSConfigOption, len(dnsoptions))
+	i := 0
+	for k, v := range dnsoptions {
+		podDnsOptions[i].Name = k
+		if v == "" {
+			podDnsOptions[i].Value = nil
+		} else {
+			podDnsOptions[i].Value = pointer.StringPtr(v)
+		}
+		i++
+	}
+	return podDnsOptions
```

Comment:
I think this could be represented by:

```suggestion
	podDnsOptions := []corev1.PodDNSConfigOption{}
	for k, v := range dnsoptions {
		podDnsOptions = append(podDnsOptions, corev1.PodDNSConfigOption{Name: k, Value: pointer.StringPtr(v)}
	}
	return podDnsOptions
```

Right?